01-make_graph.py

```diff
@@ -5,146 +5,118 @@
-        print(i, name, gh.rate_limit()['resources']['core']['remaining'])
-        r = requests.get('https://api.github.com/repos/conda-forge/'
-                         '{}-feedstock/contents/recipe/meta.yaml'.format(name),
-                         auth=(os.environ['USERNAME'], os.environ['PASSWORD']))
-        if r.status_code == 403:
-            raise github3.GitHubError(r)
-        elif r.status_code != 200:
-            print('Something odd happened to this recipe '
-                  '{}'.format(r.status_code))
-            with open('bad.txt', 'a') as f:
-                f.write('{}\n'.format(name))
-            continue
-        meta_yaml = r.json()['content']
-        if meta_yaml:
-            text = codecs.decode(b64decode(meta_yaml))
-            yaml_dict = parsed_meta_yaml(text)
-            if not yaml_dict:
-                with open('bad.txt', 'a') as f:
-                    f.write('{}\n'.format(name))
-                continue
-            # TODO: Write schema for dict
-            req = yaml_dict.get('requirements', set())
-            if req:
-                build = list(req.get('build', []) if req.get(
-                    'build', []) is not None else [])
-                run = list(req.get('run', []) if req.get(
-                    'run', []) is not None else [])
-                req = build + run
-                req = set([x.split()[0] for x in req])
-
-            if not ('url' in yaml_dict.get('source', {})
-                    and 'name' in yaml_dict.get('package', {})
-                    and 'version' in yaml_dict.get('package', {})):
-                with open('bad.txt', 'a') as f:
-                    f.write('{}\n'.format(name))
-                continue
-            sub_graph = {
-                'name': yaml_dict['package']['name'],
-                'version': str(yaml_dict['package']['version']),
-                'url': yaml_dict['source']['url'],
-                'req': req,
-                'time': time.time(),
-            }
-            k = next(iter((set(yaml_dict['source'].keys())
-                              & hashlib.algorithms_available)), None)
-            if k:
-                sub_graph['hash_type'] = k
-
-            if name in new_names:
-                gx.add_node(name, **sub_graph)
-            else:
-                gx.nodes[name].update(**sub_graph)
-            # nx.write_yaml(gx, 'graph.yml')
-            nx.write_gpickle(gx, 'graph.pkl')
-
-except github3.GitHubError as e:
-    print(e)
-    c = gh.rate_limit()['resources']['core']
-    if c['remaining'] == 0:
-        ts = c['reset']
-        print('API timeout, API returns at')
-        print(datetime.datetime.utcfromtimestamp(ts)
-              .strftime('%Y-%m-%dT%H:%M:%SZ'))
-    pass
+
+for i, name in enumerate(total_names):
+    print(i, name)
+    r = requests.get('https://raw.githubusercontent.com/conda-forge/'
+                     '{}-feedstock/master/recipe/meta.yaml'.format(name))
```

Comment:
Why not just clone the repo?

Reply:
Maybe cloning the repo is a bit slower than just requesting the `meta.yaml` and that the clone has a bunch more information that we don't need.

Reply:
We just need the meta.yaml. Faster this way